rand/rand.go

```diff
@@ -8,16 +8,10 @@ import (
 	"time"
 )
 
-var r *rand.Rand
-
 // getRandGenerator seeds a random number generator based on current time
 func getRandGenerator() *rand.Rand {
-	if r != nil {
-		return r
-	}
 	source := rand.NewSource(time.Now().UnixNano())
-	r = rand.New(source)
-	return r
+	return rand.New(source)
```

Comment:
What problem arose from using a shared `rand.Rand`? Did sharing `rand.Rand` mean that we were essentially using the same  "seed" and just getting the next number in the sequence.